gui-js/apps/minsky-electron/src/app/managers/CommandsManager.ts

```diff
@@ -58,23 +58,43 @@ export class CommandsManager {
       : classTypeRes;
 
     if (!classType) {
-      return;
+      return undefined;
     }
     return ClassType[classType];
   }
 
-  static async getItemInfo(x: number, y: number): Promise<CanvasItem> {
+  static isFalseResult(result) {
+    return !result || typeof result === 'object' && (Object.keys(result).length === 0);
+  }
 
-    if (!minsky.canvas.getItemAt(x, y)) {
+  static async getItemInfo(x: number, y: number): Promise<CanvasItem> {
+    if (this.isFalseResult(await minsky.canvas.getItemAt(x, y))) {
       return null;
     }
 
+    return await this.getCurrentItemInfo();
+  }
+
+  static async getFocusItemInfo(): Promise<CanvasItem> {
+    await minsky.canvas.setItemFromItemFocus();
+    return await this.getCurrentItemInfo();
```

Comment:
Technically true but I dislike that quirk of javascript parlance. In C# omitting await here wouldn't fly. The async tag normally turns the return type from T to Promise<T> so here it would become Promise<Promise<CanvasItem>> if not for the fact that javascript just allows the ambiguity.

Reply:
Strictly speaking true, but I dislike that quirk of javascript parlance... in C# omitting await here wouldn't fly.